pipeline/beam_tables.py

```diff
@@ -194,24 +194,13 @@ def _read_scan_text(
     A PCollection[Tuple[filename, line]] of all the lines in the files keyed
     by filename
   """
-  # List[PCollection[Tuple[filename,line]]]
-  line_pcollections_per_file: List[beam.PCollection[Tuple[str, str]]] = []
+  # PCollection[filename]
+  pfilenames = (p | beam.Create(filenames))
```

Comment:
Not for this PR, but consider creating the file list in the pipeline too.

Reply:
I think there are some speedups I can make to creating the file list. But I'm mostly not sure what creating it in the pipeline would buy us. It's not like enumerating the list is any faster on a worker machine than it is for me during pipeline setup.

One major downside to doing that is we'll lose the ability to abort creating the pipeline in the first place when there's no new data like we do [here](https://github.com/censoredplanet/censoredplanet-analysis/blob/761f614833fe8f64d771e5a70f6d088dc35285d1/pipeline/beam_tables.py#L563). We'll also lose the ability to debug the content of the file list at startup time, since it will only be available inside the pipeline.

Reply:
Listing files should be a lot faster in the worker, because it's running from inside Cloud, ideally in the same region. 

You could also parallelize the different sources.

The abort is not needed to implement: if there are no files, you won't output anything and the pipeline will end.

If you'd like to debug the list of processed files, you can save that as a side output. I did that in the past to find broken OONI files (in that case it was a list of broken files).

By the way, there are other optimization opportunities. You are currently matching against the entire dataset, even for a single date. That's bad, as it's not proportional to the work that needs to be done.

Instead, it's possible to list directories in order starting with a key. Then you can just list the files in the range you need. You'd need the GCS Client. This is what I do for OONI (see StartAfter): https://github.com/Jigsaw-Code/net-analysis/blob/2b36fe89c3305f4d1c93406725a7f46a74f246f9/netanalysis/ooni/data/ooni_client.py#L91

In any case, not for this PR.